core/src/main/java/io/questdb/cairo/StringTypeDriver.java

```diff
@@ -287,7 +290,7 @@ public long setAppendPosition(long pos, MemoryMA auxMem, MemoryMA dataMem) {
 
     @Override
     public void setColumnRefs(long address, long initialOffset, long count) {
-        Vect.setVarColumnRefs32Bit(address, initialOffset, count);
+        Vect.setVarColumnRefs32Bit(address, initialOffset, count + 1);
```

Comment:
this is a behaviour change in all other places except `WalWriter` (where you compensated for this +1)

for example here: https://github.com/jerrinot/questdb/blob/e065400768734bde734149b1272b0f2f8968f779/core/src/main/java/io/questdb/cairo/O3OpenColumnJob.java#L336

are you sure we are not trading one segfault for another? (I don't understand `O3OpenColumnJob.o3PartitionMerge()` well enough to tell without investigating how it works)

Reply:
we won't have another seg fault fortunately, but driver API is vague, i agree. It has to be more explicit about the intent.

Reply:
trying to fix here: https://github.com/questdb/questdb/pull/4362/commits/a9fea2bb81d19ab7f88efabafe82afe10d61792b